src/foam/nanos/analytics/DAOFoldManager.js

```diff
@@ -20,34 +20,45 @@ foam.CLASS({
     },
     {
       class: 'Long',
-      name: 'periodLengthMs'
+      name: 'periodLengthMs',
+      documentation: 'Convenience property for setting openTimeExpr and closeTimeExpr',
+      javaSetter: `
+setOpenTimeExpr(new foam.glang.StartOfTimeSpan.Builder(getX())
+  .setDelegate(new foam.mlang.IdentityExpr.Builder(getX()).build())
+  .setTimeSpanMs(val)
+  .build());
+setCloseTimeExpr(new foam.glang.EndOfTimeSpan.Builder(getX())
+  .setDelegate(new foam.mlang.IdentityExpr.Builder(getX()).build())
+  .setTimeSpanMs(val)
+  .build());
+      `
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'openTimeExpr'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'closeTimeExpr'
     }
   ],
   methods: [
     {
       name: 'foldForState',
-      // TODO: use a more efficient key based lock
```

Comment:
Is this TODO no longer relevant?  The method is still synchronized.

Reply:
My bad, didn't mean to remove it. Started looking at actually addressing the TODO but stepped back because I don't have time to really get into it at the moment.

Reply:
I removed/improved the synchronization.

Reply:
Yup so this comment can finally go :)
Thanks Kevin!